Added GUI docker support

rushic24 opened the pull request:
i added a Dockerfile for setting up this repo. Some guides recommend venv, some virtualenv some conda, and all of them mostly don't work the same way. I created this dockerfile to run everything inside docker enviornment, so just using a few docker build and docker run command should get you this environment working. 

I'm not sure about the location for documentation, since the complete repository is empty as of now. 
I have added how do use the dockerfile under ` ## Using Docker` section in the main README.md.

To bring GUI outside this docker you need to forward X like for arch linux its

`xhost +` command. This will change depending on the OS, so feel free to add to it if you know about other OSes. 

![image](https://user-images.githubusercontent.com/6279035/147395069-7dedd1e8-9626-43f3-8adc-e1f86fb087ad.png)

sveneschlbeck commented:
@rushic24 Thanks for your interest, Could you roughly explain what you did and why you did it. Otherwise, people who want to understand it (including me) have to go through all commits to check the work.

It would be great to here some words concerning:
1. What was done?
2. Why was it done?
3. How does it differ from the solutions prior to the commits?
4. How is ist or shall it be documented?

rushic24 commented:
Sure,

In this commit I added a Dockerfile for setting up this repo. Some guides recommend venv, some virtualenv some conda, and all of them mostly don't work the same way. I created this dockerfile to run everything inside docker enviornment, so just using a few docker build and docker run command should get you this environment working. 

I'm not sure about the location for documentation, since the complete repository is empty as of now. 
I have added how do use the dockerfile under ` ## Using Docker` section in the main README.md.

To bring GUI outside this docker you need to forward X like for arch linux its

`xhost +` command. This will change depending on the OS, so feel free to add to it if you know about other OSes.

sveneschlbeck commented:
@rushic24 Thank you, currently some multi-language implementations have priority (and they could change some dependencies), so Docker will have to wait for some time, will notify you if I get on it
